metadata/src/main/java/org/apache/kafka/metadata/migration/MigrationManifest.java

```diff
@@ -41,15 +41,17 @@ public static class Builder {
         private final Map<MetadataRecordType, Integer> counts = new HashMap<>();
         private int batches = 0;
         private int total = 0;
+        private long batchDurationsNs = 0;
         private long endTimeNanos = 0;
 
         Builder(Time time) {
             this.time = time;
             this.startTimeNanos = time.nanoseconds();
         }
 
-        public void acceptBatch(List<ApiMessageAndVersion> recordBatch) {
+        public void acceptBatch(List<ApiMessageAndVersion> recordBatch, long durationNs) {
             batches++;
+            batchDurationsNs += durationNs;
```

Comment:
Okay. This is measure how much time the ZK migration spent in the controller, writing and committing the batches to KRaft, right?

Reply:
Yup, that's correct.